src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -265,7 +265,10 @@ MPL_STATIC_INLINE_PREFIX uint32_t MPIDI_OFI_winfo_disp_unit(MPIR_Win * win, int
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_WINFO_DISP_UNIT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_WINFO_DISP_UNIT);
 
-    if (MPIDI_OFI_WIN(win).winfo)
+    /* Always use winfo if any of PROV_KEY and VIRT_ADDRESS is on.
+     * It avoids the checking instruction for winfo  */
+    if (MPIDI_OFI_ENABLE_MR_PROV_KEY || MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS ||
+        MPIDI_OFI_WIN(win).winfo)
         ret = MPIDI_OFI_WIN(win).winfo[rank].disp_unit;
     else
         ret = win->disp_unit;
```

Comment:
The per-target displacement unit is already the slow branch, so I am not sure it worth the optimization here -- we are going to derefence `.wininfo` array already, what are we saving here?

Reply:
This is not the slow path. For scalable MR (! MPIDI_OFI_ENABLE_MR_PROV_KEY && ! MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS), we don't need load `MPIDI_OFI_WIN(win).winfo` at all. But without this patch, it causes 1 load instruction to check.

Reply:
You only skip the check when `MPIDI_OFI_ENABLE_MR_PROV_KEY || MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS`, not `! MPIDI_OFI_ENABLE_MR_PROV_KEY && ! MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS`, or did I read it wrong?

Reply:
> For basic MR (MPIDI_OFI_ENABLE_MR_PROV_KEY || MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS), it causes 1 test instruction which is unnecessary. We always use winfo in such case.

I don't  think that one instruction is worth optimizing, since we gonna load the cacheline anyway

Reply:
I changed my response. Sorry my previous response did not make sense.

Reply:
I do need it for the openshmem work. We need explain every instruction in the put/get path. This 1 instruction is useless, so how can we explain it? :-)